### PR TITLE
simple docker setup

### PR DESCRIPTION
**Pull request by rpshukla**
Get a simple docker setup with x11 forwarding and volumes

### CODE REVIEW

**Comment by rpshukla** [comment]
I get the following errors when I try to run gui applications:

```
libGL error: MESA-LOADER: failed to retrieve device information
libGL error: Version 4 or later of flush extension not found
libGL error: failed to load driver: i915
libGL error: failed to open drm device: No such file or directory
libGL error: failed to load driver: i965
```

Maybe I need to install additional drivers in the docker container?

**Comment by rpshukla** [comment]
Also, you need to set `ROS_IP` or else ros will complain that it can't contact its own master when you roslaunch things.

To fix this, run this inside the docker container:

```
export ROS_IP=127.0.0.1
```

**Comment by rpshukla** [comment]
Update: gazebo now opens but gzserver crashes with the following message:

```
gzserver: /build/ogre-1.9-mqY1wq/ogre-1.9-1.9.0+dfsg1/OgreMain/src/OgreRenderSystem.cpp:546: virtual void Ogre::RenderSystem::setDepthBufferFor(Ogre::RenderTarget*): Assertion `b
Attached && "A new DepthBuffer for a RenderTarget was created, but after creation" "it says it's incompatible with that RT"' failed.
Aborted (core dumped)
Traceback (most recent call last):
  File "/opt/ros/kinetic/lib/controller_manager/spawner", line 207, in <module>
    if __name__ == '__main__': main()
  File "/opt/ros/kinetic/lib/controller_manager/spawner", line 185, in main
    resp = load_controller(name)
  File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/impl/tcpros_service.py", line 435, in __call__
    return self.call(*args, **kwds)
  File "/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/impl/tcpros_service.py", line 505, in call
    raise ServiceException("unable to connect to service: %s"%e)
rospy.service.ServiceException: unable to connect to service: [Errno 104] Connection reset by peer
[WARN] [1568215136.048747, 783.769000]: Controller Spawner error while taking down controllers: unable to connect to service: [Errno 111] Connection refused
[gazebo-2] process has died [pid 116, exit code 134, cmd /opt/ros/kinetic/lib/gazebo_ros/gzserver -e ode /root/ros/src/spear_simulator/worlds/ball_find_00.world __name:=gazebo __
log:=/root/.ros/log/6e67e550-d4a7-11e9-affc-9cb6d08ceb8f/gazebo-2.log].
log file: /root/.ros/log/6e67e550-d4a7-11e9-affc-9cb6d08ceb8f/gazebo-2*.log
```

**Comment by rpshukla** [comment]
Update: I fixed gzserver crashing on my system by adding `--device=/dev/dri:/dev/dri` to the docker run script.

See: http://wiki.ros.org/docker/Tutorials/Hardware%20Acceleration

**Comment by rpshukla** [comment]
The next thing I have to do it get volumes working for the working directory

File: README.md [hunk 1]
```diff
@@ -28,7 +28,109 @@ This should ideally be able to run on the TX2, but it is not necessary for it to
...
 
-# Setup and install instructions
+# Docker setup and install instructions
+
+## Requirements
+
+- Any linux distro
+- Docker
+
+Some distributions have old versions of docker. The recommended way to install
```
Comment (Petelliott) [inline README.md]: Do we actually need that new a version of docker? the travis build seems to work and in uses Xenial.

File: README.md [hunk 2]
```diff
@@ -28,7 +28,109 @@ This should ideally be able to run on the TX2, but it is not necessary for it to
...
+
+More information about linux configuration of docker can be found here:
+https://docs.docker.com/install/linux/linux-postinstall/
+
+## Build Docker Image
+
+First pull the base ROS docker image:
+
+``` bash
+docker pull ros:kinetic-robot
```
Comment (Petelliott) [inline README.md]: This should be automatically pulled when you build the image

File: README.md [hunk 3]
```diff
@@ -28,7 +28,109 @@ This should ideally be able to run on the TX2, but it is not necessary for it to
...
+```
+
+To build our specific docker image, run these commands in the same directory as `Dockerfile`.
+
+Note: these 2 commands may take several minutes.
+
+``` bash
+docker build -t spear-env - < spear-env.Dockerfile
```
Comment (Petelliott) [inline README.md]: `docker build -t spear-env -f spear-env.Dockerfile .`

will this actually work without specifiyng the build environment as `.`?

spear-env will be removed in the future

Comment (rpshukla) [reply README.md]: This works. I'll leave it for now and change it when spear-env gets removed

**Pull request merged**